internal/monitors/conviva/configsetter.go

```diff
@@ -0,0 +1,153 @@
+package conviva
+
+import (
+	"context"
+	"strconv"
+	"strings"
+)
+
+func setConfigFields(m *Monitor, conf *Config)  {
+	conf.metriclensMetricNames = map[string][]string{
+		"quality_metriclens": {
+			"total_attempts",
+			"video_start_failures_percent",
+			"exits_before_video_start_percent",
+			"plays_percent",
+			"video_startup_time_sec",
+			"rebuffering_ratio_percent",
+			"average_bitrate_kbps",
+			"video_playback_failures_percent",
+			"ended_plays",
+			"connection_induced_rebuffering_ratio_percent",
+			"video_restart_time",
+		},
+		"audience_metriclens": {
+			"concurrent_plays",
+			"plays",
+			"ended_plays",
+		},
+	}
+	m.ctx, m.cancel = context.WithCancel(context.Background())
+	if conf.MetricConfigs == nil {
+		conf.MetricConfigs = []*MetricConfig{{Metric: "quality_metriclens"}}
+	}
+	setAccounts(m, conf)
+	setFilters(m, conf)
+	setDimensions(m, conf)
+	setMetriclensFilters(m, conf)
+}
+
+func setAccounts(m *Monitor, conf *Config)  {
+	jsonResponse, err := get(m, conf, "https://api.conviva.com/insights/2.4/accounts.json")
+	if err != nil {
+		logger.Errorf("Get accounts request failed %v\n", err)
```

Comment:
Propagate the errors back to the calling function all the way back to the main Configure method on the monitor so that errors are clearer.

Also, for robustness, I would make it do this configuration on the first interval before it collects metrics in case the conviva API server is down temporarily when the agent starts up.  Have a check for whether it has been dynamically configured, and if not, have it do it before fetching metric values.  That way, if the Conviva server is down for an extended period, this monitor will start working after it comes back up automatically without user intervention.